Packs/MicrosoftGraphSecurity/Integrations/MicrosoftGraphSecurity/MicrosoftGraphSecurity.py

```diff
@@ -321,6 +321,90 @@ def list_threat_assessment_requests(self, filters=None, order_by=None, sort_orde
         return self.ms_client.http_request(method='GET', url_suffix=THREAT_ASSESSMENT_URL_PREFIX,
                                            params=params, retries=1, status_list_to_retry=[405])
 
+    def advanced_hunting_request(self, query: str, timeout: int):
+        """
+        POST request to the advanced hunting API:
+        Args:
+            query (str): query advanced hunting query language
+            timeout (int): The amount of time (in seconds) that a request will wait for a client to
+                     establish a connection to a remote machine before a timeout occurs.
+
+        Returns:
+            The response object contains three top-level properties:
+
+                Stats - A dictionary of query performance statistics.
+                Schema - The schema of the response, a list of Name-Type pairs for each column.
+                Results - A list of advanced hunting events.
+        """
+        return self.ms_client.http_request(method='POST', url_suffix='security/runHuntingQuery', json_data={"Query": query},
+                                           timeout=timeout)
+
+    def get_incidents_request(self, incident_id: Optional[int], limit: int, timeout: int, args_for_filter: dict) -> dict:
+        """
+        GET request to get single incident.
+        Args:
+            incident_id (int): incident's id
+            timeout (int): waiting time for command execution
+
+
+       Returns( Dict): request results as dict:
+                    { '@odata.context',
+                      'value': updated incident,
+                    }
+
+        """
+
+        if incident_id:
+            url_suffix = f'security/incidents/{incident_id}'
+        else:
+            url_suffix = 'security/incidents?$count=true'
+            if limit:
+                url_suffix += f'&$top={str(limit)}'
+            if args_for_filter['status']:
```

Comment:
Could we use a safe access in all of those cases? Are we sure that those keys exist?

Reply:
It's a dictionary I create in the previous function, so I'm sure it will arrive